File: src/codeforces/contest1329/ProblemA.java
package codeforces.contest1329;

import java.util.Scanner;

public class ProblemA {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        int[] l = new int[m];
        for (int i = 0; i < m; i++) {
            l[i] = scanner.nextInt();
            if (i + l[i] > n) {
                System.out.println(-1);
                return;
            }
        }
        int[] ans = new int[m];
        long s = 0;
        int left = 0;
        for (int i = 0; i < m; i++) {
            s += l[i];
            ans[i] = i;
            left = Math.max(left, i + l[i]);
        }
        if (s < n) {
            System.out.println(-1);
            return;
        }
        if (left != n) {
            int i = m - 1;
            left = n;
            while (ans[i] + l[i] < left) {
                ans[i] = Math.max(left - l[i], i);
                left = ans[i];
                i--;
            }
        }
        for (int a : ans) {
            System.out.println(a + 1);
        }
    }
}

File: src/codeforces/contest873/ProblemA.kt
@file:Suppress("DuplicatedCode")

package codeforces.contest873

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.util.*

private const val p = 1_000_000_007L;

fun main(args: Array<String>) {
    val t = nextInt()
    for (i in 0 until t) {
        val n = nextInt()
        println(solve(nextInts(n), nextInts(n)))
    }

}

fun solve(a: IntArray, b: IntArray): Long {
//    a.shuffle()
//    b.shuffle()
    a.sort()
    b.sort()
    val n = a.size

    var ans = 1L
    var j = 0
    for (i in 0 until n) {
        while (j < n && a[i] > b[j]) j++
        if (i >= j) return 0L
        ans = ans * (j - i) % p
    }
    return ans
}


private class MyReader(inputStream: InputStream) {
    private val reader = BufferedReader(InputStreamReader(inputStream))
    private var tokenizer: StringTokenizer? = null

    fun next(): String {
        while (true) {
            tokenizer.let {
                if (it == null || !it.hasMoreTokens()) {
                    tokenizer = StringTokenizer(reader.readLine())
                } else {
                    return it.nextToken()
                }
            }
        }
    }
}

private val reader = MyReader(System.`in`)
private fun next() = reader.next()
private fun nextInt() = next().toInt()
private fun nextLong() = next().toLong()
private fun nextDouble() = next().toDouble()
private fun nextStrings(n: Int) = Array(n) { next() }
private fun nextInts(n: Int) = IntArray(n) { nextInt() }
private fun nextLongs(n: Int) = LongArray(n) { nextLong() }
private fun nextDoubles(n: Int) = DoubleArray(n) { nextDouble() }